internal/provider/resources/deployment.go

```diff
@@ -432,16 +444,43 @@ func mapPullStepsTerraformToAPI(tfPullSteps []PullStepModel) ([]api.PullStep, di
 	for i := range tfPullSteps {
 		tfPullStep := tfPullSteps[i]
 
-		apiPullStep := api.PullStep{
-			Type:        tfPullStep.Type.ValueString(),
+		pullStepCommon := api.PullStepCommon{
 			Credentials: tfPullStep.Credentials.ValueStringPointer(),
 			Requires:    tfPullStep.Requires.ValueStringPointer(),
-			Directory:   tfPullStep.Directory.ValueStringPointer(),
-			Repository:  tfPullStep.Repository.ValueStringPointer(),
-			Branch:      tfPullStep.Branch.ValueStringPointer(),
-			AccessToken: tfPullStep.AccessToken.ValueStringPointer(),
-			Bucket:      tfPullStep.Bucket.ValueStringPointer(),
-			Folder:      tfPullStep.Folder.ValueStringPointer(),
+		}
+
+		// Steps that pull from remote storage have the same fields.
+		// Define the struct here for reuse in each of those cases.
+		pullStepPullFrom := api.PullStepPullFrom{
+			PullStepCommon: pullStepCommon,
+			Bucket:         tfPullStep.Bucket.ValueStringPointer(),
+			Folder:         tfPullStep.Folder.ValueStringPointer(),
+		}
+
+		var apiPullStep api.PullStep
+		switch tfPullStep.Type.ValueString() {
+		case "git_clone":
+			apiPullStep.PullStepGitClone = &api.PullStepGitClone{
+				PullStepCommon:    pullStepCommon,
+				Repository:        tfPullStep.Repository.ValueStringPointer(),
+				Branch:            tfPullStep.Branch.ValueStringPointer(),
+				AccessToken:       tfPullStep.AccessToken.ValueStringPointer(),
+				IncludeSubmodules: tfPullStep.IncludeSubmodules.ValueBoolPointer(),
+			}
```

Comment:
This is obviously lengthier than the logic we defined before, but because of the payload expected by the API, I have to put configuration under certain fields in the parent struct so that I can give that parent struct the correct JSON struct tag like `prefect.deployments.steps.git_clone`.